src/wilson/dom.cljs

```diff
@@ -174,9 +174,15 @@
   (update elem 1 merge-attrs attrs))
 
 (defn with-class
-  "Adds a class to an element."
-  [cls elem]
-  (with-attrs {:class cls} elem))
+  "Adds a class to a component."
+  [cls component]
+  (if (vector? component)
```

Comment:
Technically I care if it's a fn in the second case, cause then what I do is completely different. I don't see how match helps me here? If it's a vector (with or without map of attrs) then it's a hiccup, right? I don't need to know more